sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormatSuite.scala

```diff
@@ -26,6 +26,22 @@ import org.apache.spark.sql.test.SharedSQLContext
 
 class ParquetFileFormatSuite extends QueryTest with ParquetTest with SharedSQLContext {
 
+  test("Number of threads doesn't grow extremely after parquet file reading") {
+    withTempDir { dir =>
+      val file = dir.toString + "/file"
+      spark.range(1).toDF("a").coalesce(1).write.parquet(file)
+      spark.read.parquet(file)
+      val numThreadBefore = Thread.activeCount
+      (1 to 100).map { _ =>
+        spark.read.parquet(file)
+      }
+      val numThreadAfter = Thread.activeCount
+      // Hard to test a correct thread number,
+      // but it shouldn't increase more than a reasonable number.
+      assert(numThreadAfter - numThreadBefore < 20)
```

Comment:
It reduces to few after waiting an enough time. The number returned by Thread.activeCount is only an estimate. So we may not expect this to be 0.